File: src/gateway/routes/logs.ts
import { Hono } from "hono";
import type { GatewayOptions } from "../types.js";

export function logRoutes(options: GatewayOptions): Hono {
  const app = new Hono();

  app.get("/", async (c) => {
    if (!options.storage) return c.json({ error: "storage not configured" }, 501);
    const sessionId = c.req.query("sessionId");
    const limit = parseInt(c.req.query("limit") ?? "100", 10);
    const logs = await options.storage.listLogs(sessionId, limit);
    return c.json(logs);
  });

  return app;
}

File: titanx/gateway/routes/logs.py
from __future__ import annotations

from fastapi import APIRouter
from fastapi.responses import JSONResponse

from ..types import GatewayOptions


def logs_router(options: GatewayOptions) -> APIRouter:
    router = APIRouter()

    @router.get("")
    async def list_logs(sessionId: str | None = None, limit: int = 100):
        if not options.storage:
            return JSONResponse({"error": "storage not configured"}, status_code=501)
        return await options.storage.list_logs(sessionId, limit)

    return router
